pkg/templates/metricsserver/deployment.go

```diff
@@ -208,6 +208,7 @@ func metricsServerDeployment() *appsv1.Deployment {
 							ImagePullPolicy: corev1.PullIfNotPresent,
 							Args: []string{
 								"--kubelet-insecure-tls",
+								"--kubelet-preferred-address-types=InternalIP,InternalDNS,ExternalDNS,ExternalIP",
```

Comment:
metrics-server will use first accessible address type.

on hetzner it will be `ExternalIP`, others do support `InternalIP`. With provider `none` real external IP will be indicated as `InternalIP` so will work ether way.